yt_dlp/extractor/twitch.py

```diff
@@ -380,13 +380,16 @@ def _download_info(self, item_id):
             }],
             'Downloading stream metadata GraphQL')
 
-        video = traverse_obj(data, (0, 'data', 'video'))
-        video['moments'] = traverse_obj(data, (1, 'data', 'video', 'moments', 'edges', ..., 'node'))
-        video['storyboard'] = traverse_obj(data, (2, 'data', 'video', 'seekPreviewsURL'), expected_type=url_or_none)
+        initial_index = 1 if data[0].get('errors') else 0
 
+        video = traverse_obj(data, (initial_index, 'data', 'video'))
         if video is None:
-            raise ExtractorError(
-                'Video %s does not exist' % item_id, expected=True)
+            raise ExtractorError(f'Video {item_id} does not exist', expected=True)
+
+        video['moments'] = traverse_obj(data, (initial_index + 1, 'data', 'video', 'moments', 'edges', ..., 'node'))
+        video['storyboard'] = traverse_obj(
+            data, (initial_index + 2, 'data', 'video', 'seekPreviewsURL'), expected_type=url_or_none)
```

Comment:
```suggestion
        video = traverse_obj(data, (..., 'data', 'video'))
        if not video:
            raise ExtractorError(f'Video {item_id} does not exist', expected=True)

        video['moments'] = traverse_obj(data, (..., 'data', 'video', 'moments', 'edges', ..., 'node'))
        video['storyboard'] = traverse_obj(data, (..., 'data', 'video', 'seekPreviewsURL',{url_or_none}), get_all=False)

```
Wouldn't this work?

Reply:
Mmm it doesn't

The bug is quite rare, but to debug this I load the json I posted above

```python
f = open('/home/amish/Documents/Programming/yt-dlp/tmp/twitch-error.json')
data = json.load(f)

video = traverse_obj(data, (..., 'data', 'video'))
...
```

```bash
[debug] Command-line config: ['https://www.twitch.tv/videos/1536751224', '--verbose', '-F']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.02.17 [a0a7c0154] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 5d75e21fa
[debug] Python 3.10.7 (CPython x86_64 64bit) - Linux-5.19.0-31-generic-x86_64-with-glibc2.36 (OpenSSL 3.0.5 5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg N-108931-g4dda3b1653-20221104 (setts), ffprobe N-108931-g4dda3b1653-20221104, phantomjs broken, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
[twitch:vod] Extracting URL: https://www.twitch.tv/videos/1536751224
[twitch:vod] 1536751224: Downloading stream metadata GraphQL
ERROR: list indices must be integers or slices, not str
Traceback (most recent call last):
  File "/home/amish/Documents/Programming/yt-dlp/yt_dlp/YoutubeDL.py", line 1502, in wrapper
    return func(self, *args, **kwargs)
  File "/home/amish/Documents/Programming/yt-dlp/yt_dlp/YoutubeDL.py", line 1578, in __extract_info
    ie_result = ie.extract(url)
  File "/home/amish/Documents/Programming/yt-dlp/yt_dlp/extractor/common.py", line 693, in extract
    ie_result = self._real_extract(url)
  File "/home/amish/Documents/Programming/yt-dlp/yt_dlp/extractor/twitch.py", line 520, in _real_extract
    video = self._download_info(vod_id)
  File "/home/amish/Documents/Programming/yt-dlp/yt_dlp/extractor/twitch.py", line 390, in _download_info
    video['moments'] = traverse_obj(data, (..., 'data', 'video', 'moments', 'edges', ..., 'node'))
TypeError: list indices must be integers or slices, not str
```

Reply:
mb. It needs get_all=False
```diff
-        video = traverse_obj(data, (..., 'data', 'video'))
+        video = traverse_obj(data, (..., 'data', 'video'), get_all=False)
```
Edited above comment accordingly

Reply:
`...` with `get_all=False` looks at all indices, returning the first one. So unless the json can contain unrelated videos, this would be better than checking just 0/1 index